pkg/gds/models/v1/contacts_test.go

```diff
@@ -0,0 +1,145 @@
+package models_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/trisacrypto/directory/pkg/gds/models/v1"
+	pb "github.com/trisacrypto/trisa/pkg/trisa/gds/models/v1beta1"
+)
+
+func TestIterContacts(t *testing.T) {
+	contacts := &pb.Contacts{
+		Technical: &pb.Contact{
+			Name: "technical",
+		},
+		Administrative: &pb.Contact{
+			Email: "administrative@example.com",
+		},
+		Billing: &pb.Contact{
+			Name: "billing",
+		},
+		Legal: &pb.Contact{
+			Email: "legal@example.com",
+		},
+	}
+	expectedContacts := []*pb.Contact{
+		contacts.Technical,
+		contacts.Administrative,
+		contacts.Legal,
+		contacts.Billing,
+	}
+	expectedKinds := []string{
+		models.TechnicalContact,
+		models.AdministrativeContact,
+		models.LegalContact,
+		models.BillingContact,
+	}
+
+	actualContacts := []*pb.Contact{}
+	actualKinds := []string{}
+
+	// Should iterate over all contacts.
+	iter := models.NewContactIterator(contacts, false, false)
+	for iter.Next() {
+		contact, kind := iter.Value()
+		actualContacts = append(actualContacts, contact)
+		actualKinds = append(actualKinds, kind)
+	}
+	require.NoError(t, iter.Error())
+	require.Equal(t, expectedContacts, actualContacts)
+	require.Equal(t, expectedKinds, actualKinds)
+
+	actualContacts = []*pb.Contact{}
+	actualKinds = []string{}
+
+	// Should skip contacts without an email address.
+	expectedContacts = []*pb.Contact{
+		contacts.Administrative,
+		contacts.Legal,
+	}
+	expectedKinds = []string{
+		models.AdministrativeContact,
+		models.LegalContact,
+	}
+	iter = models.NewContactIterator(contacts, true, false)
+	for iter.Next() {
+		contact, kind := iter.Value()
+		actualContacts = append(actualContacts, contact)
+		actualKinds = append(actualKinds, kind)
+	}
+	require.NoError(t, iter.Error())
+	require.Equal(t, expectedContacts, actualContacts)
+	require.Equal(t, expectedKinds, actualKinds)
+
+	actualContacts = []*pb.Contact{}
+	actualKinds = []string{}
+
+	// Should skip nil contacts.
+	contacts.Technical = nil
+	contacts.Billing = nil
+	iter = models.NewContactIterator(contacts, false, false)
+	for iter.Next() {
+		contact, kind := iter.Value()
+		actualContacts = append(actualContacts, contact)
+		actualKinds = append(actualKinds, kind)
+	}
+	require.NoError(t, iter.Error())
+	require.Equal(t, expectedContacts, actualContacts)
+	require.Equal(t, expectedKinds, actualKinds)
+}
+
+func TestIterVerfiiedContacts(t *testing.T) {
```

Comment:
```suggestion
func TestIterVerifiedContacts(t *testing.T) {
```

Reply:
Agreed, the error handling was a bit of a headache but I'm pretty happy with how this turned out!